source/beamline/CLS/CLSJJSlit.cpp

```diff
@@ -1,128 +1,55 @@
-//	statusPVControl_ = new AMReadOnlyPVControl(name+" Status", pvBaseName+":status", this);
-	gapPVControl_ = new AMPVControl(name+" Gap", pvBaseName+":gap:fbk", pvBaseName+":gap", QString(), this, tolerance, moveStartTimeoutSeconds);
-	centerPVControl_ = new AMPVControl(name+" Center", pvBaseName+":center:fbk", pvBaseName+":center", QString(), this, tolerance, moveStartTimeoutSeconds);
-	statusPVControl_ = new AMReadOnlyPVControl(name+" Status", pvBaseName+":status", this);
-
-	connect(gapPVControl_, SIGNAL(connected(bool)), this, SLOT(onPVControlConnected()));
-	connect(gapPVControl_, SIGNAL(valueChanged(double)), this, SIGNAL(gapValueChanged(double)));
-	connect(centerPVControl_, SIGNAL(connected(bool)), this, SLOT(onPVControlConnected()));
-	connect(centerPVControl_, SIGNAL(valueChanged(double)), this, SIGNAL(centerValueChanged(double)));
-	connect(statusPVControl_, SIGNAL(connected(bool)), this, SLOT(onPVControlConnected()));
-	connect(statusPVControl_, SIGNAL(valueChanged(double)), this, SLOT(onStatusValueChanged()));
-
-	wasConnected_ = false;
-	onPVControlConnected();
-}
-
-bool CLSJJSlitBladesControl::isConnected() const
-{
-	return gapPVControl_->isConnected() && centerPVControl_->isConnected() && statusPVControl_->isConnected();
-}
-
-void CLSJJSlitBladesControl::onPVControlConnected()
-{
-	bool connectionStatus = isConnected();
-
-	if (wasConnected_ != connectionStatus) {
-		wasConnected_ = connectionStatus;
-		emit connected(wasConnected_);
-	}
-}
+	// Initialize member variables.
 
-void CLSJJSlitBladesControl::moveGap(double setpoint)
-{
-	gapPVControl_->move(setpoint);
-}
-
-void CLSJJSlitBladesControl::moveCenter(double setpoint)
-{
-	centerPVControl_->move(setpoint);
-}
-
-void CLSJJSlitBladesControl::onStatusValueChanged()
-{
-	emit statusStringChanged(statusString());
-}
-
-// ======================== implementation for CLSJJSlit ======================================
-CLSJJSlit::~CLSJJSlit()
-{
-
-}
-
-CLSJJSlit::CLSJJSlit(const QString &name, const QString &description, const QString &verticalBladesPVBaseName, const QString &horizontalBladesPVBaseName, double tolerance, double moveStartTimeoutSeconds, double limit, QObject *parent):
-	QObject(parent)
-{
 	name_ = name;
-	description_ = description;
-	limit_ = limit;
+	connected_ = false;
 
-	verticalBladesControl_ = new CLSJJSlitBladesControl(name+" Vertical", description+" Vertical", CLSJJSlitBladesControl::Vertical, verticalBladesPVBaseName, tolerance, moveStartTimeoutSeconds, this);
-	horizontalBladesControl_ = new CLSJJSlitBladesControl(name+" Horizontal", description+" Horizontal", CLSJJSlitBladesControl::Horizontal, horizontalBladesPVBaseName, tolerance, moveStartTimeoutSeconds, this);
+	upperBlade_ = new AMPVwStatusControl(name_+"UpperBlade", upperBladePVName+":mm:fbk", upperBladePVName+":mm", upperBladePVName+":status", upperBladePVName+":stop", this, 0.005);
```

Comment:
Based on the conversation with @beaured0, you will probably want to change this from ```:mm:fbk``` to ```:mm:sp```.